NimbleMediumTests/Sources/Specs/Data/Repositories/ArticleCommentRepositorySpec.swift

```diff
@@ -0,0 +1,81 @@
+//
+//  ArticleCommentRepositorySpec.swift
+//  NimbleMediumTests
+//
+//  Created by Mark G on 15/09/2021.
+//
+
+import Quick
+import Nimble
+import RxNimble
+import RxSwift
+import RxTest
+
+@testable import NimbleMedium
+
+final class ArticleCommentRepositorySpec: QuickSpec {
+
+    override func spec() {
+        var repository: ArticleCommentRepository!
+        var networkAPI: NetworkAPIProtocolMock!
+        var scheduler: TestScheduler!
+        var disposeBag: DisposeBag!
+
+        describe("an ArticleCommentRepository") {
+
+            beforeEach {
+                disposeBag = DisposeBag()
+                networkAPI = NetworkAPIProtocolMock()
+                repository = ArticleCommentRepository(networkAPI: networkAPI)
+                scheduler = TestScheduler(initialClock: 0)
+            }
+
+            describe("its getComments() call") {
+
+                context("when the request returns success") {
+
+                    var outputArticleComments: TestableObserver<[APIArticleComment]>!
+                    let inputResponse = APIArticleCommentsResponse.dummy
+
+                    beforeEach {
+                        outputArticleComments = scheduler.createObserver([APIArticleComment].self)
+                        networkAPI.setPerformRequestForReturnValue(Single.just(inputResponse))
+                        repository.getComments(slug: "")
+                        .asObservable()
+                        .map {
+                            $0.compactMap { $0 as? APIArticleComment }
+                        }
```

Comment:
can one line this I think, it doesn't exceed the max line length constraint yet 😁
```suggestion
                        .map { $0.compactMap { $0 as? APIArticleComment } }
```